src/is_intersection_empty.jl

```diff
@@ -1365,3 +1365,82 @@ function is_intersection_empty(X::LazySet{N},
                                   {N<:Real}
     return is_intersection_empty(C, X, witness)
 end
+
+"""
+    is_intersection_empty(X::CartesianProductArray{N, S},
+                          Y::HPolyhedron{N}) where {N<:Real, S<:LazySet{N}}
+
+Check whether a Cartesian product array intersect with a H-polyhedron.
+
+### Input
+
+- `X` -- Cartesian product array of convex sets
+- `Y` -- H-polyhedron (can be unbounded)
+
+### Output
+
+`true` iff ``X ∩ Y = ∅ ``.
+"""
+function is_intersection_empty(X::CartesianProductArray{N, S},
+                               Y::HPolyhedron{N}) where {N<:Real, S<:LazySet{N}}
+    if isbounded(Y)
+        # no free variables
+        blocks = block_indices(X)
+    else
+        constrained_vars = constrained_dimensions(Y)
+        blocks = block_indices(X, constrained_vars)
+    end
+
+    result = CartesianProductArray(length(X.array), N)
+
+    low_set = CartesianProductArray(length(blocks), N)
```

Comment:
As in the other PR, better create the `Vector{<:S}(undef, length(blocks))` here and create the CPA after the loop.